pkg/minikube/machine/client.go

```diff
@@ -18,23 +18,176 @@ package machine
 
 import (
 	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net"
 	"os"
+	"path/filepath"
 	"time"
 
-	"net"
+	"k8s.io/minikube/pkg/minikube/constants"
 
-	"github.com/docker/machine/drivers/hyperv"
 	"github.com/docker/machine/drivers/virtualbox"
-	"github.com/docker/machine/drivers/vmwarefusion"
+	"github.com/docker/machine/libmachine"
 	"github.com/docker/machine/libmachine/auth"
 	"github.com/docker/machine/libmachine/cert"
 	"github.com/docker/machine/libmachine/check"
-	"github.com/docker/machine/libmachine/drivers/plugin"
+	"github.com/docker/machine/libmachine/drivers"
 	"github.com/docker/machine/libmachine/drivers/plugin/localbinary"
+	"github.com/docker/machine/libmachine/drivers/rpc"
+	"github.com/docker/machine/libmachine/engine"
 	"github.com/docker/machine/libmachine/host"
+	"github.com/docker/machine/libmachine/persist"
+	"github.com/docker/machine/libmachine/swarm"
+	"github.com/docker/machine/libmachine/version"
 	"github.com/golang/glog"
+	"github.com/pkg/errors"
 )
 
+type driverGetter func(string, []byte) (drivers.Driver, error)
+
+type ClientType int
+type clientFactory interface {
+	NewClient(string, string) libmachine.API
+}
+
+type localClientFactory struct{}
+
+func (*localClientFactory) NewClient(storePath, certsDir string) libmachine.API {
+	return &LocalClient{
+		certsDir:  certsDir,
+		storePath: storePath,
+		Filestore: persist.NewFilestore(storePath, certsDir, certsDir),
+	}
+}
+
+type rpcClientFactory struct{}
+
+func (*rpcClientFactory) NewClient(storePath, certsDir string) libmachine.API {
+	return libmachine.NewClient(storePath, certsDir)
+}
+
+var clientFactories = map[ClientType]clientFactory{
+	ClientTypeLocal: &localClientFactory{},
+	ClientTypeRPC:   &rpcClientFactory{},
+}
+
+const (
+	ClientTypeLocal ClientType = iota
+	ClientTypeRPC
+)
+
+// Gets a new client depending on the clientType specified
+// defaults to the libmachine client
+func NewAPIClient(clientType ClientType) libmachine.API {
+	storePath := constants.Minipath
+	certsDir := constants.MakeMiniPath("certs")
+	newClientFactory, ok := clientFactories[clientType]
+	if !ok {
+		glog.Warningf("No implementation for API client type %d, trying RPC implementation", clientType)
```

Comment:
Ah, I remember why I did this.  This method also doesn't return an error.  Do you think I should change the signature to return `(libmachine.API, error)` and change all its cmd package callers to check for an error?